src/fondant/pipeline/compiler.py

```diff
@@ -515,3 +517,167 @@ def _set_configuration(self, task, fondant_component_operation):
             task.set_accelerator_type(accelerator_name)
 
         return task
+
+
+class SagemakerCompiler(Compiler):
+    def __init__(self):
+        self._resolve_imports()
+
+    def _resolve_imports(self):
+        try:
+            import sagemaker
+            import sagemaker.processing
+            import sagemaker.workflow.pipeline
+            import sagemaker.workflow.steps
+
+            self.sagemaker = sagemaker
+
+        except ImportError:
+            msg = """You need to install the sagemaker extras to use the sagemaker compiler,\n
+                     you can install it with `pip install fondant[sagemaker]`"""
+            raise ImportError(
+                msg,
+            )
+
+    def build_command(
+        self,
+        metadata: Metadata,
+        arguments: t.Dict[str, t.Any],
+        dependencies: t.List[str] = [],
+    ) -> t.List[str]:
+        # add metadata argument to command
+        command = ["--metadata", f"'{metadata.to_json()}'"]
+
+        # add in and out manifest paths to command
+        command.extend(
+            [
+                "--output_manifest_path",
+                f"{metadata.base_path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                f"{metadata.component_id}/manifest.json",
+            ],
+        )
+
+        # add arguments if any to command
+        for key, value in arguments.items():
+            if isinstance(value, (dict, list)):
+                command.extend([f"--{key}", f"'{json.dumps(value)}'"])
+            else:
+                command.extend([f"--{key}", f"'{value}'"])
+
+        # resolve dependencies
+        if dependencies:
+            for dependency in dependencies:
+                # there is only an input manifest if the component has dependencies
+                command.extend(
+                    [
+                        "--input_manifest_path",
+                        f"{metadata.base_path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                        f"{dependency}/manifest.json",
+                    ],
+                )
+
+        return command
+
+    def compile(
+        self,
+        pipeline: Pipeline,
+        output_path: str,
+        instance_type: str = "ml.t3.medium",
```

Comment:
Could we use a different machine type? I think you had to wait a while until the provisioning was done?

Reply:
I still need to do some experimentation here, but I will update it once I have a better view.